### PR TITLE
Disable gateway PDB by default and add description

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by hanxiaop** [comment]
This is a follow-up of https://github.com/istio/istio/pull/44481, which addresses comments in https://github.com/istio/istio/pull/44481#pullrequestreview-1413295627 and https://github.com/istio/istio/pull/44481#pullrequestreview-1413367828

File: manifests/charts/gateway/values.yaml [hunk 1]
```diff
@@ -96,5 +96,32 @@ imagePullPolicy: ""
...
+# Or, to allow a maximum of 1 unavailable replica, you can set:
+#
+# podDisruptionBudget:
+#   maxUnavailable: 1
+#
+# You can also specify the `unhealthyPodEvictionPolicy` field, and the valid values are `IfHealthyBudget` and `AlwaysAllow`.
+# For example, to set the `unhealthyPodEvictionPolicy` to `AlwaysAllow`, you can update this value as follows:
+#
+# podDisruptionBudget:
+#   minAvailable: 1
```
Comment (howardjohn) [inline manifests/charts/gateway/values.yaml:120]: this chart has a values schema - can this even be set or will helm reject it?

Comment (howardjohn) [reply manifests/charts/gateway/values.yaml:120]: actually you may have handled that in the original PR, my bad

**Review by howardjohn — APPROVED** [review]

File: manifests/charts/gateway/values.yaml (hunk 1 above)
Comment (hanxiaop) [reply manifests/charts/gateway/values.yaml:120]: Yes this is already handled in values schema. `minAvailable: 1` and `minAvailable: 1%` are both acceptable format.

File: manifests/charts/gateway/values.yaml [hunk 2]
```diff
@@ -96,5 +96,32 @@ imagePullPolicy: ""
 
 imagePullSecrets: []
 
-podDisruptionBudget:
-  minAvailable: 1
+# This value is used to configure a Kubernetes PodDisruptionBudget for the gateway.
+#
+# By default, the `podDisruptionBudget` is disabled (set to `{}`),
```
Comment (costinm) [inline manifests/charts/gateway/values.yaml:101]: That's a nice summary of the PDB doc - I was looking more for 'the value is a fragment of [link to PDB doc]' and 
'in a test/small/dev environment it is safe to leave it off, in production we recommend setting it to X'. 

I don't actually know what is the right X - some examples would help if someone has run this in prod.
Does it depend on cluster size ( a large ingress with ~10 replicas would use 1 as a value or 4 ? ). 
Do we recommend or not unhealtyPodEvictionPolicy for ingress ? 

That's the real problem - we need to give some operational advice to users. I'm a developer not an SRE - so 
people who run this in prod are much more qualified.

We probably need to have this on the wiki or doc site too - but most users just look at the helm params,
AFAIK the docs show up.

**Review by costinm — APPROVED** [review]
Good improvement - would be great if we can make it more actionable, but this PR is already moving things forward, would be nice to follow up with the other undocumented fields.

**Pull request merged**